cdap-common/src/main/java/co/cask/cdap/common/MethodNotAllowedException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common;
+
+import org.jboss.netty.handler.codec.http.HttpMethod;
+
+/**
+ * Thrown when some method is not allowed.
+ */
+public class MethodNotAllowedException extends Exception {
+
+  private HttpMethod method;
```

Comment:
would be good to add getters for these. Also, do we need the first constructor on line 29? If not, lets remove it and make these members final? We can add it if required later. But ideally, there shouldn't be a MethodNotALlowedException without a method and an apiPath.